net/kadnode/files/kadnode.init

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh /etc/rc.common
+
+START=95
+USE_PROCD=1
+KADNODE_BIN=/usr/bin/kadnode
+CONFIG_FILE=/tmp/kadnode.conf
+OPTS=""
+
+
+xappend() {
+	local name="$2" value="$1"
+	OPTS="$OPTS--${name//_/-} ${value//'/\\'}
+"
+}
+
+append_opts_list() {
+	local name cfg="$1"; shift
+	for name in $*; do
+		config_list_foreach "$cfg" "$name" xappend "$name"
+	done
+}
+
+append_opts() {
+	local name value cfg="$1"; shift
+	for name in $*; do
+		config_get value "$cfg" "$name"
+		[ -n "$value" ] && xappend "$value" "$name"
+	done
+}
+
+append_opts_boolean() {
+	local name value cfg="$1"; shift
+	for name in $*; do
+		config_get_bool value "$cfg" "$name" 0
+		[ $value -gt 0 ] && xappend '' $name
+	done
+}
+
+section_enabled() {
+	config_get_bool enabled "$1" 'enabled' 0
+	[ $enabled -gt 0 ]
+}
+
+first_cfg() {
+	local cfg
+	lookup() {
+		[ -z "$cfg" ] && cfg=$1
+	}
+	config_foreach lookup 'kadnode'
+	echo $cfg
+}
+
+start_service() {
+	config_load 'kadnode'
+	local cfg="$(first_cfg)"
```

Comment:
are you explicitly allowing there to be multiple UCI configs, but only startinng the first one?  Why not let people do crazy things like have two.  it doesn't cost you anything, and actually means you have _less_ code?

Reply:
I do not see any use case for running multiple instances. That is why I take the first section only. It also simplifies the code (I would need to handle specific /tmp/kadnode.conf files).